### ISSUE TITLE
Rule 'no-use-before-define' from eslint

### ISSUE THREAD

**Issue by alkorlos** [sveltejs/eslint-plugin-svelte#674]
### Description

Hello, thanks for the great tool

Eslint have rule ['no-use-before-define'](https://eslint.org/docs/latest/rules/no-use-before-define), it is useful in many situations, but have problem with Svelte:

`+layout.svelte`
```html
<Header />

<script>
	import Header from '$lib/components/header/Header.svelte';
</script>
```
Will result: `error  'Header' was used before it was defined  no-use-before-define`. This is correct behavior to JS, but not to Svelte.

Is there any way to improve compatibility?

**Comment by DMartens** [comment]
Also ran into this.
When contributing this rule, should it support coarse options extending the existing options like `props` and `components` or just allow any reference outside the <script> as that would be the only difference between the new rule and the core rule?

**Comment by baseballyama** [comment]
The original rule uses a scope manager, so I’m not sure how to extend the rule for this.🤔

**Comment by ota-meshi** [comment]
I don't think the rule is giving a false positive.
The following script causes the rule to report `<Header />`, but I don't think it's a false positive.

```js
/*eslint no-use-before-define: ["error"]*/

function render() {
  return <Header />
}

import Header from '$lib/components/header/Header.svelte';
```

I think you should either turn it off or write `<script>` first.

[Demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdXNlLWJlZm9yZS1kZWZpbmU6IFtcImVycm9yXCJdKi9cblxuZnVuY3Rpb24gcmVuZGVyKCkge1xuICByZXR1cm4gPEhlYWRlciAvPlxufVxuXG5pbXBvcnQgSGVhZGVyIGZyb20gJyRsaWIvY29tcG9uZW50cy9oZWFkZXIvSGVhZGVyLnN2ZWx0ZSc7Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fX19fQ==)
